simulators/portal-bridge-tests/src/main.rs

```diff
@@ -0,0 +1,180 @@
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt::init();
+
+    let mut suite = Suite {
+        name: "portal-bridge-tests".to_string(),
+        description: "The portal bridge test suite".to_string(),
+        tests: vec![],
+    };
+
+    suite.add(TestSpec {
+        name: "Portal bridge tests".to_string(),
+        description: "".to_string(),
+        always_run: false,
+        run: test_portal_bridge,
+        client: None,
+    });
+
+    let sim = Simulation::new();
+    run_suite(sim, suite).await;
+}
+
+async fn run_suite(host: Simulation, suite: Suite) {
+    let name = suite.clone().name;
+    let description = suite.clone().description;
+
+    let suite_id = host.start_suite(name, description, "".to_string()).await;
+
+    for test in &suite.tests {
+        test.run_test(host.clone(), suite_id, suite.clone()).await;
+    }
+
+    host.end_suite(suite_id).await;
+}
+
+dyn_async! {
+   async fn test_portal_bridge<'a> (test: &'a mut Test, _client: Option<Client>) {
+        // Get all available portal clients
+        let clients = test.sim.client_types().await;
+
+        let private_key_1 = "fc34e57cc83ed45aae140152fd84e2c21d1f4d46e19452e13acc7ee90daa5bac".to_string();
+        let private_key_2 = "e5add57dc4c9ef382509e61ce106ec86f60eb73bbfe326b00f54bf8e1819ba11".to_string();
+        let private_key_3 = "80449579bd9d09a03d245b0a018f36887a0e97d3d2994d89841e9f5bd5bfedf0".to_string();
+
+        // Iterate over all possible pairings of clients and run the tests (including self-pairings)
+        for ((client_a, client_b), client_c) in clients.iter().cartesian_product(clients.iter()).cartesian_product(clients.iter()) {
+            test.run(
+                NClientTestSpec {
+                    name: format!("Bridge test. A:{} --> B:{} --> C:{}", client_a.name, client_b.name, client_c.name),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_bridge,
+                    private_keys: Some(vec![Some(private_key_1.clone()), Some(private_key_2.clone()), Some(private_key_3.clone())]),
+                    clients: vec![client_a.clone(), client_b.clone(), client_c.clone()],
+                }
+            ).await;
+        }
+   }
+}
+
+dyn_async! {
+    async fn test_bridge<'a> (clients: Vec<Client>) {
+        let (client_a, client_b, client_c) = match clients.iter().collect_tuple() {
+            Some((client_a, client_b, client_c)) => (client_a, client_b, client_c),
+            None => {
+                panic!("Unable to get expected amount of clients from NClientTestSpec");
+            }
+        };
+
+        let master_acc = MasterAccumulator::default();
+        let header_oracle = HeaderOracle::new(master_acc);
+        let portal_clients = vec![client_a.rpc.clone()];
+        let epoch_acc_path = "validation_assets/epoch_acc.bin".into();
+        let mode = BridgeMode::Test("./test-data/test_data_collection_of_forks_blocks.json".into());
```

Comment:
So if I understand it right, this is a mode where you feed the Trin Bridge with content from these files, and the Trin bridge connects to node a, and gossips the data into the network via node a.

If that's right, this test could also just be repeated in another simulator (portal-mesh?) without the bridge, and by iterating over the content and calling `portal_historyGossip` on 1 of the clients/nodes?

Reply:
We can yes you are totally right

Reply:
The point of this group of tests is to test the bridge, more than the clients, right? So taking the bridge out of the equation would not really accomplish the goal...

But maybe we're saying there's a separate non-bridge test that is worth bringing over to mesh, independently?

Reply:
That is why I made this PR https://github.com/ethereum/portal-hive/pull/97 . I couldn't think of a reason how the 1 to 1 gossip test would need 3 nodes, wait maybe it is cause we would see if node B re-gossips the data to node C. Well I didn't think of it that way. So if we want to test if our code re-gossips data it got gossiped to it. I can see the use in that yeah.

Reply:
Yeah, it wasn't fully clear to me whether the point here was to test the bridge or to do an A -> B -> C gossip test.

Because for testing the bridge you could just have an A -> B test. And for testing Gossip you could have the A -> B -> C test in the other simulator.

Anyway, it's clear now to me that this is simulator for testing bridge functionality. And that you can use it without having pandaops access. So, its all good to me :+1:

Reply:
I can quickly change it to a A -> B test.

@carver do you have any blocking concerns with this PR or do you think it is fine if I merge this after I address kdeme's comments?

Reply:
> I can quickly change it to a A -> B test.

I'm fine with either.

Reply:
I generally agree with KIm that if we want to test the bridge, we don't need to run N clients test, A -> B would be enough here. Similar to the gossip, if we want to test the gossip, we don't need to include the bridge.